tonic/src/transport/tls.rs

```diff
@@ -15,6 +15,11 @@ impl Certificate {
     /// Parse a PEM encoded X509 Certificate.
     ///
     /// The provided PEM should include at least one PEM encoded certificate.
+    ///
+    /// ## SAFETY
+    /// `pem` must contain valid PEM encoded certificate. It is undefined
+    /// behavior to call this with `pem` that is not valid PEM encoded
```

Comment:
Is this really undefined behavior? When I hear "UB" I think things like "the compiler is free to do whatever it wants", but in this case it seems like the behavior is pretty clear: You get a valid (in the rust sense) Certificate object containing invalid data (in the business logic sense). Am I missing something?

Reply:
Thanks! Makes sense. The "UB" should be in the meaning of Rust. I updated it just a detailed document. By the way, now that I think about it, the constructor name is `from_pem` so it might not as confusing as I thought it would be. If this note section seems unnecessary, I will remove it.